test/podFailover/workload/main.go

```diff
@@ -17,49 +17,175 @@ limitations under the License.
 package main
 
 import (
+	"context"
 	"crypto/tls"
 	"crypto/x509"
+	"encoding/json"
 	"flag"
+	"fmt"
 	"log"
 	"net/http"
 	"net/url"
 	"os"
+	"path/filepath"
 	"strconv"
 	"time"
 
+	"github.com/Azure/azure-amqp-common-go/v3/aad"
+	eventhub "github.com/Azure/azure-event-hubs-go/v3"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
 	"k8s.io/klog/v2"
+	podfailure "sigs.k8s.io/azuredisk-csi-driver/test/podFailover/podFailoverCrd"
 )
 
-var mountPath = flag.String("mount-path", "", "The path of the file where timestamps will be logged")
+var mountPath = flag.String("mount-path", "/", "The path of the file where timestamps will be logged")
 var metricsEndpoint = flag.String("metrics-endpoint", "", "The endpoint where prometheus metrics should be published")
 var authEnabled = flag.Bool("auth-enabled", false, "Specify whether request to metrics endpoint requires authentication or not")
-var testName = flag.String("test-name", "default-test-run", "The name of the test to be used as metrics label to distinguish metrics sent from multiple test runs")
+var runID = flag.String("run-id", "default", "The run id of the test")
+var storageType = flag.String("storage-type", "", "The underlying storage type of test")
+var workloadType = flag.String("workload-type", "default-workload-type", "The type of test being ran(1pod1pvc, 3pod3pvc, etc)")
+var driverVersion = flag.String("driver-version", "", "The version of csi-driver")
+var namespace = flag.String("namespace", "pod-failover", "The namespace resources are created in")
+
+type DowntimeMetric struct {
+	TimeStamp     string `json:"timeStamp"`
+	RunID         string `json:"runID"`
+	StorageType   string `json:"storageType"`
+	DriverVersion string `json:"driverVersion"`
+	ImageVersion  string `json:"imageVersion"`
+	FailureType   string `json:"failureType"`
+	WorkloadType  string `json:"workloadType"`
+	PodDowntime   int64  `json:"downtime"`
+}
 
 func main() {
 
 	flag.Parse()
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		kubeConfigPath := os.Getenv("KUBECONFIG")
+		if len(kubeConfigPath) == 0 {
+			kubeConfigPath = filepath.Join(os.Getenv("HOME"), ".kube", "config")
+		}
+		config, _ = clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+	}
+
+	clientset, _ := kubernetes.NewForConfig(config)
+
+	podFailoverClient, err := podfailure.NewForConfig(config)
+	if err != nil {
+		klog.Errorf("Error occurred while getting pod failure client: %v", err)
+	}
+
+	ctx := context.Background()
+
+	podName := os.Getenv("MY_POD_NAME")
```

Comment:
Why by environment variable instead of command-line parameter? Should we fail if this variable is not set, i.e. returns ""?

Reply:
this will always be set. because stateful sets generate with a number at the end. I needed a way to get the pod name. I can get the generated pod name by 
env:
  - name: MY_POD_NAME
    valueFrom:
      fieldRef:
        fieldPath: metadata.name

Reply:
Makes sense.